Jantimizer/QueryTestSuite/TestRunners/TestCase.cs

```diff
@@ -0,0 +1,24 @@
+﻿using QueryTestSuite.Models;
+using CsvHelper;
+
+namespace QueryTestSuite.TestRunners
+{
+    internal class TestCase
+    {
+        public string Name { get; private set; }
+        public string Category { get; private set; }
+        public AnalysisResult AnalysisResult { get; private set; }
+
+        public TestCase (string Name, string Category)
+        {
+            this.Name = Name;
+            this.Category = Category;
+        }
+
+        public async Task<TestCase> Run(Task<AnalysisResult> AnalysisResult)
+        {
+            this.AnalysisResult = await AnalysisResult;
```

Comment:
Please dont name a input parameter the same as the type! Also odd casing for the parameter in the first place. Wouldnt it, by convention, be camel case for function parameters?